### DISCUSSION TITLE
Check if using S3 or Local Storage

### DISCUSSION THREAD

**Discussion by manishonc** [pocketbase/pocketbase#2978] [Q&A] [answered]
I am a little bit confused. A while back, I set up S3 with Pocketbase, and the file URL was an S3 URL. But now, I have set up S3, and the URL is my domain name for the file.

I am confused; am I using AWS S3 or not? It says it's connected to S3, and whenever I am uploading something, it appears in the S3 bucket.

**Comment by ganigeorgiev** [comment]
I don't understand what do you mean.

If the uploaded file appear in the S3 bucket, then you are using the configured S3 storage.

**Reply by manishonc** [reply]
When I upload a file in my Pocketbase, I get the URL: https://upp.examplus.co.in/api/files/2hieklubg9tbahh/cbadjjti676sh7t/hindi_1973909_960_720_VR52zHnU8C.webp?token=
This is what I am receiving in the API request, and it's working.

However, I want to use Cloudfront, and when I use this method:
https://yourcdn.com/yourCollectionId/yourRecordId/yourFile.mp4

It doesn't seem to work, but when I make the object public in S3, Cloudfront works.

Interestingly, the URL I am getting in Pocketbase is working without making it public.

How do I set up Cloudfront without making my whole bucket public? If I were getting an S3 URL in Pocketbase, I would simply change the CDN URL, and that would work.

What should I do?

**Reply by manishonc** [reply]
It's just that I can't use S3 without Cloudfront because S3 is So Slow for our content .

**Reply by ganigeorgiev** [reply, marked as answer]
> How do I set up Cloudfront without making my whole bucket public?

If you want to use a CDN endpoint you have to make the bucket public. ACL per object is not supported because not all S3 vendors have this implemented.

> Interestingly, the URL I am getting in Pocketbase is working without making it public.

That's because PocketBase is using the S3 APIs with the configured credentials to access the file.

**Reply by ganigeorgiev** [reply]
Depending on your use case and your collections structure, if you don't want to make the entire bucket public, you can also create a S3 bucket policy in your AWS console (_if you use AWS_) and whitelist the objects that has "directory"/prefix your collection id(s).

**Reply by manishonc** [reply]
> Depending on your use case and your collections structure, if you don't want to make the entire bucket public, you can also create a S3 bucket policy in your AWS console (_if you use AWS_) and whitelist the objects that has "directory"/prefix your collection id(s).

Thanks
